homeassistant/components/sensor/twitch.py

```diff
@@ -12,38 +12,48 @@
 from homeassistant.helpers.entity import Entity
 import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['python-twitch==1.3.0']
+
+REQUIREMENTS = ['python-twitch-client==0.5.1']
 
 _LOGGER = logging.getLogger(__name__)
 
 ATTR_GAME = 'game'
 ATTR_TITLE = 'title'
 
 CONF_CHANNELS = 'channels'
+CONF_CLIENTID = 'clientid'
 ICON = 'mdi:twitch'
 
 STATE_OFFLINE = 'offline'
 STATE_STREAMING = 'streaming'
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CLIENTID): cv.string,
     vol.Required(CONF_CHANNELS, default=[]):
         vol.All(cv.ensure_list, [cv.string]),
 })
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Twitch platform."""
+    from twitch import TwitchClient
+
     channels = config.get(CONF_CHANNELS, [])
+    client = TwitchClient(client_id=config.get(CONF_CLIENTID))
```

Comment:
What happens if the client ID is not valid?

Reply:
the API normally responds with a HTTP 400 error if this is incorrect or missing, I have done this to do a quick test to see if the key is right or the api is even working